adobe_vipm/flows/sync/agreement.py

```diff
@@ -982,3 +869,332 @@ def _check_adobe_deployment_id(deployment_id: str, adobe_deployment: dict) -> bo
 
 def _is_subscription_in_set(subscription_ids: set, subscription: dict) -> bool:
     return subscription["subscriptionId"] in subscription_ids
+
+
+def sync_agreements_by_3yc_end_date(
+    mpt_client: MPTClient, adobe_client: AdobeClient, *, dry_run: bool
+) -> None:
+    """
+    Synchronizes agreements by their active subscriptions renewed yesterday.
+
+    Args:
+        adobe_client: Adobe Client
+        mpt_client: MPT API client.
+        dry_run: Run in dry run mode.
+    """
+    logger.info("Syncing agreements by 3yc End Date...")
+    _sync_agreements_by_param(
+        mpt_client, adobe_client, Param.THREE_YC_END_DATE.value, dry_run=dry_run, sync_prices=True
+    )
+
+
+def sync_agreements_by_coterm_date(
+    mpt_client: MPTClient, adobe_client: AdobeClient, *, dry_run: bool
+) -> None:
+    """
+    Synchronizes agreements by their active subscriptions renewed yesterday.
+
+    Args:
+        adobe_client: Adobe API client.
+        mpt_client: MPT API client.
+        dry_run: Run in dry run mode.
+    """
+    logger.info("Synchronizing agreements by cotermDate...")
+    _sync_agreements_by_param(
+        mpt_client, adobe_client, Param.COTERM_DATE.value, dry_run=dry_run, sync_prices=True
+    )
+
+
+def _sync_agreements_by_param(
+    mpt_client: MPTClient,
+    adobe_client: AdobeClient,
+    param: Param,
+    *,
+    dry_run: bool,
+    sync_prices: bool,
+) -> None:
+    today = dt.datetime.now(tz=dt.UTC).date()
+    today_iso = today.isoformat()
+    yesterday = (today - dt.timedelta(days=1)).isoformat()
+    rql_query = (
+        "eq(status,Active)&"
+        f"any(parameters.fulfillment,and(eq(externalId,{param}),eq(displayValue,{yesterday})))&"
+        f"any(parameters.fulfillment,and(eq(externalId,{Param.LAST_SYNC_DATE.value}),ne(displayValue,{today_iso})))&"
+        # Let's get only what we need
+        "select=lines,parameters,assets,subscriptions,product,listing"
+    )
+    for agreement in mpt.get_agreements_by_query(mpt_client, rql_query):
+        sync_agreement(
+            mpt_client, adobe_client, agreement, dry_run=dry_run, sync_prices=sync_prices
+        )
+
+
+def sync_agreements_by_renewal_date(
+    mpt_client: MPTClient, adobe_client: AdobeClient, *, dry_run: bool
+) -> None:
+    """
+    Synchronizes agreements by their active subscriptions renewed yesterday.
+
+    Args:
+        adobe_client: Adobe API client used for API operations.
+        mpt_client: MPT API client.
+        dry_run: Run in dry run mode.
+    """
+    logger.info("Synchronizing agreements by renewal date...")
+    today_plus_1_year = dt.datetime.now(tz=dt.UTC).date() + relativedelta(years=1)
+    today_iso = dt.datetime.now(tz=dt.UTC).date().isoformat()
+    yesterday_every_month = (
+        (today_plus_1_year - dt.timedelta(days=1) - relativedelta(months=month)).isoformat()
+        for month in range(24)
+    )
+
+    rql_query = (
```

Comment:
And here also - pls, filter by MPT_PRODUCT_IDS that we have in configuration/settings

Reply:
Done